src/include/dhandle.h

```diff
@@ -33,6 +33,10 @@
 	(F_ISSET(dhandle, WT_DHANDLE_DEAD) ||				\
 	!F_ISSET(dhandle, WT_DHANDLE_EXCLUSIVE | WT_DHANDLE_OPEN))
 
+/* Cursor's data handle. */
+#define	WT_CURSOR_DHANDLE(c)						\
+	(((WT_CURSOR_BTREE *)(c))->btree->dhandle)
+
```

Comment:
I'd prefer a macro called `WT_SESSION_META_DHANDLE` that takes a session and looks up the data handle `session->meta_cursor` to find the data handle for this one case.  This only works on btree cursors (so the name is misleading) and is only ever used on `session->meta_cursor`.